pallets/polkadex/src/tests.rs

```diff
@@ -49,35 +49,44 @@ fn check_trading_engine() {
+        assert_ok!(DEXModule::submit_order(Origin::signed(bob),AskLimit,trading_pair,10000*UNIT,(1*UNIT)/10));
         // Place some random market orders
-        assert_ok!(DEXModule::submit_order(Origin::signed(alice),BidMarket,trading_pair,(UNIT/100)*5,0));
-        assert_ok!(DEXModule::submit_order(Origin::signed(bob),AskMarket,trading_pair,0,(UNIT/1000)*5));
-        assert_ok!(DEXModule::submit_order(Origin::signed(alice),BidMarket,trading_pair,(UNIT/1000)*16,0));
-        assert_ok!(DEXModule::submit_order(Origin::signed(bob),AskMarket,trading_pair,0,(UNIT/1000)*16));
+        assert_ok!(DEXModule::submit_order(Origin::signed(alice),BidMarket,trading_pair,500*UNIT,0));
+        assert_ok!(DEXModule::submit_order(Origin::signed(bob),AskMarket,trading_pair,0,(UNIT/100)*5));
+        assert_ok!(DEXModule::submit_order(Origin::signed(alice),BidMarket,trading_pair,1646*UNIT,0));
+        assert_ok!(DEXModule::submit_order(Origin::signed(bob),AskMarket,trading_pair,0,(UNIT/100)*16));
+        // Partial limit orders for Alice ( Token1 ) and Bob ( Token 2)
+        assert_ok!(DEXModule::submit_order(Origin::signed(alice),BidLimit,trading_pair,10600*UNIT,(5*UNIT)/100));
+        assert_ok!(DEXModule::submit_order(Origin::signed(bob),AskLimit,trading_pair,8400*UNIT,(5*UNIT)/100));
+        // Full+half queue limit orders for Alice ( Token1 ) and Bob ( Token 2)
+        assert_ok!(DEXModule::submit_order(Origin::signed(alice),BidLimit,trading_pair,10750*UNIT,(14*UNIT)/100));
+        assert_ok!(DEXModule::submit_order(Origin::signed(bob),AskLimit,trading_pair,8200*UNIT,(14*UNIT)/100));
         // Read the block chain state for verifying
         // Balances of Token #1 for Alice
-        assert_eq!(pallet_generic_asset::Module::<Test>::free_balance(&1, &alice), ((UNIT / 1000) * 496934));
-        assert_eq!(pallet_generic_asset::Module::<Test>::reserved_balance(&1, &alice), ((UNIT / 10) * 4841));
+        // If buyer protection enabled, Token #1 free balance for Alice = 795
+        assert_eq!(pallet_generic_asset::Module::<Test>::free_balance(&1, &alice), ((UNIT / 1000) * 789000));
+        assert_eq!(pallet_generic_asset::Module::<Test>::reserved_balance(&1, &alice), ((UNIT / 1000) * 1620000));
         // Balances of Token #2 for Alice
-        assert_eq!(pallet_generic_asset::Module::<Test>::free_balance(&2, &alice), ((UNIT / 1000000) * 21066));
+        assert_eq!(pallet_generic_asset::Module::<Test>::free_balance(&2, &alice), (80*UNIT)/100);
         assert_eq!(pallet_generic_asset::Module::<Test>::reserved_balance(&2, &alice), 0);
         // Balances of Token #1 for Bob
-        assert_eq!(pallet_generic_asset::Module::<Test>::free_balance(&1, &bob), ((UNIT / 1000) * 18966));
+        // If buyer protection enabled, Token #1 free balance for Bob = 7585
+        assert_eq!(pallet_generic_asset::Module::<Test>::free_balance(&1, &bob), 7591*UNIT);
```

Comment:
As I was enabled Buyer Protection in the engine, can you change the 7591 to 7585 and commit again.

Reply:
ok